components/script/dom/mutationrecord.rs

```diff
@@ -18,6 +23,47 @@ pub struct MutationRecord {
 
     //property for target node
     target: JS<Node>,
+
+    //property for attribute name
+    attribute_name: DOMRefCell<Option<DOMString>>,
+
+    //property for attribute namespace
+    attribute_namespace: DOMRefCell<Option<DOMString>>,
+
+    //property for old value
+    old_value: DOMRefCell<Option<DOMString>>,
+}
+
+impl MutationRecord {
+    pub fn new(record_type: DOMString, target: &Node) -> Root<MutationRecord> {
+        let boxed_record = box MutationRecord::new_inherited(record_type, target);
+        return reflect_dom_object(boxed_record, window_from_node(target).deref(),
+            MutationRecordBinding::Wrap);
+    }
+
+    fn new_inherited(record_type: DOMString, target: &Node) -> MutationRecord {
+        MutationRecord {
+            reflector_: Reflector::new(),
+            record_type: record_type,
+            target: JS::from_ref(target),
+            attribute_name: Default::default(),
+            attribute_namespace: Default::default(),
+            old_value: Default::default(),
+        }
+    }
+
+    // setter for attr_name
+    pub fn SetAttributeName(&self, attr_name: DOMString) {
+        *self.attribute_name.borrow_mut() = Some(attr_name);
+    }
+    // setter for attr_namespace
+    pub fn SetAttributeNamespace(&self, attr_namespace: DOMString) {
+        *self.attribute_name.borrow_mut() = Some(attr_namespace);
+    }
+    // setter for attr_oldvalue
+    pub fn SetoldValue(&self, attr_oldvalue: DOMString) {
+        *self.attribute_name.borrow_mut() = Some(attr_oldvalue);
+    }
```

Comment:
These three setters are overwriting the same member instead of setting separate members.